Source/ACE/Managers/WorldManager.cs

```diff
@@ -10,14 +10,17 @@
 using ACE.Entity;
 using ACE.Network;
 
+using log4net;
+
 namespace ACE.Managers
 {
     public static class WorldManager
     {
-        // commented: unused.  uncomment if you'll use it.
-        // private static uint sessionTimeout = 150u; // max time between packets before the client disconnects
+        private static readonly ILog log = LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
 
-        private static readonly List<Session> sessionStore = new List<Session>();
+        public const ushort ServerId = 0xB;
+        private static Session[] sessionMap = new Session[128]; // TODO Placeholder, should be config MaxSessions

```

Comment:
During login, the client can be assigned a unique Id, which is then sent with every packet that client sends.  The array ensures we are allocating available ID's and reserving them for that specific session.  We in turn leverage that to find sessions by a single int array lookup for each packet processed, instead of iterating the full session list doing IpEndPoint comparison.